Test/Unit/Visitor/Arithmetic.php

```diff
@@ -223,4 +223,66 @@ public function case_change_default_context()
                     ->call('getVariable')->withArguments($variableName)->once
                     ->call('getConstant')->withArguments('PI')->once;
     }
+
+    public function case_visitor_hexadecimal_number()
+    {
+        $this
+            ->given(
+                $compiler           = Compiler\Llk\Llk::load(new File\Read('hoa://Library/Math/Arithmetic.pp')),
+                $visitor            = new CUT(),
+                $hexadecimalNumber  = '0x2a'
+            )
+            ->then
+                ->object($compiler->parse($hexadecimalNumber))
+                    ->isInstanceOf('Hoa\Compiler\Llk\TreeNode')
```

Comment:
You can check the ID and the value of the token (resp. `token` and `hexadecimal`).
Same for subsequent test cases.